torch/_dynamo/side_effects.py

```diff
@@ -211,7 +211,14 @@ def track_object_new(
     ):
         obj = object_new(user_cls)
         variable = variable_cls(
-            obj, mutable_local=AttributeMutationNew(None, cls_source), **options
+            obj,
+            mutable_local=AttributeMutationNew(None, cls_source),
+            # TODO(whc) this is a hack, i'm not sure what to do.
```

Comment:
@voznesenskym @ezyang any idea here?

Reply:
ok i figured this out (partly)

1) cls_source is definitely wrong here, since that represents the class not the instance, and that's just wrong.  passing cls_source is the cause of the `event_dim` problem TODO'd in guards.py in this PR.  So at least both mysteries are solved.

2) what can we do instead?  (A) is to not pass a source here, and then make carveouts in a bunch of places where we allow source to be None and propagate None in those cases.  TBD whether this really works or runs into trouble.  (B) is to come up with the notion of an 'InlinedVariableSource' that somehow ties in with the side effect machinery so that if we need to guard on something we can codegen the temporary variable.

@voznesenskym any idea about whether (B) is a good option?